src/main/java/org/waarp/openr66/context/ErrorCode.java

```diff
@@ -30,216 +30,155 @@
     /**
      * Code stands for Size not allowed
      */
-    SizeNotAllowed(Messages.getString("ErrorCode.33"), 'd'); //$NON-NLS-1$
+    SizeNotAllowed('d'); 
 
-    /**
-     * Literal for this code
-     */
-    public String mesg;
     /**
      * Code could be used to switch case operations
      */
     public char code;
 
-    private ErrorCode(String mesg, char code) {
-        this.mesg = mesg;
+    private ErrorCode(char code) {
         this.code = code;
     }
 
     public String getCode() {
         return String.valueOf(code);
     }
 
-    /**
-     * Update messages from current Language
-     */
-    public static void updateLang() {
-        InitOk.mesg = Messages.getString("ErrorCode.0");
-        PreProcessingOk.mesg = Messages.getString("ErrorCode.1");
-        TransferOk.mesg = Messages.getString("ErrorCode.2");
-        PostProcessingOk.mesg = Messages.getString("ErrorCode.3");
-        CompleteOk.mesg = Messages.getString("ErrorCode.4");
-        ConnectionImpossible.mesg = Messages.getString("ErrorCode.5");
-        ServerOverloaded.mesg = Messages.getString("ErrorCode.6");
-        BadAuthent.mesg = Messages.getString("ErrorCode.7");
-        ExternalOp.mesg = Messages.getString("ErrorCode.8");
-        TransferError.mesg = Messages.getString("ErrorCode.9");
-        MD5Error.mesg = Messages.getString("ErrorCode.10");
-        Disconnection.mesg = Messages.getString("ErrorCode.11");
-        RemoteShutdown.mesg = Messages.getString("ErrorCode.12");
-        FinalOp.mesg = Messages.getString("ErrorCode.13");
-        Unimplemented.mesg = Messages.getString("ErrorCode.14");
-        Shutdown.mesg = Messages.getString("ErrorCode.15");
-        RemoteError.mesg = Messages.getString("ErrorCode.16");
-        Internal.mesg = Messages.getString("ErrorCode.17");
-        StoppedTransfer.mesg = Messages.getString("ErrorCode.18");
-        CanceledTransfer.mesg = Messages.getString("ErrorCode.19");
-        Warning.mesg = Messages.getString("ErrorCode.20");
-        Unknown.mesg = Messages.getString("ErrorCode.21");
-        QueryAlreadyFinished.mesg = Messages.getString("ErrorCode.22");
-        QueryStillRunning.mesg = Messages.getString("ErrorCode.23");
-        NotKnownHost.mesg = Messages.getString("ErrorCode.24");
-        LoopSelfRequestedHost.mesg = Messages.getString("ErrorCode.25");
-        QueryRemotelyUnknown.mesg = Messages.getString("ErrorCode.26");
-        FileNotFound.mesg = Messages.getString("ErrorCode.27");
-        CommandNotFound.mesg = Messages.getString("ErrorCode.28");
-        PassThroughMode.mesg = Messages.getString("ErrorCode.29");
-        Running.mesg = Messages.getString("ErrorCode.30");
-        IncorrectCommand.mesg = Messages.getString("ErrorCode.31");
-        FileNotAllowed.mesg = Messages.getString("ErrorCode.32");
-        SizeNotAllowed.mesg = Messages.getString("ErrorCode.33");
-    }
+    public String getMesg() { return Messages.getString("ErrorCode." + code); }
```

Comment:
Elegant but active code instead of passive ones (previous code mas statically assigned so saving extra query to Messages. However, Messages being a Map, it should be ok.

Reply:
I tested it it works on both  monitoring and admin interfaces.
It's easier to use because it doesn't require the ErrorCode static Map to be initialized which is hard to handle in unit testing.
Moreover it gives a direct relationship between the message code and the error id.

Reply:
I understand ;-)